In [10]:
import cv2
import os
import pandas as pd
from tqdm import tqdm

In [4]:
df = pd.read_csv("../data/list_attr_celeba.csv").loc[:, ["image_id", "Smiling"]]

In [5]:
df.shape

(202599, 2)

In [6]:
df["Smiling"].value_counts()

Smiling
-1    104930
 1     97669
Name: count, dtype: int64

In [7]:
img_dir = "../data/img_align_celeba/img_align_celeba"

In [11]:
shapes = {}

for img_name in tqdm(df["image_id"]):
    img_path = os.path.join(img_dir, img_name)
    img = cv2.imread(img_path)
    shape = (img.shape[1], img.shape[0])
    shapes[shape] = shapes.get(shape, 0) + 1

for shape, number in shapes.items():
    print(shape, number)

100%|██████████| 202599/202599 [00:59<00:00, 3389.63it/s]

(178, 218) 202599


Conclusions:
* Overall, we have 202559 images. 
* Smiling attribute has two values, -1 and 1. Value -1 has 114930 samples whereas value 1 has 97669. We don't have the problem of unbalanced sampling.
* All images have a size of 178 by 218.